File: src/main/java/tk/hiddenname/smarthome/model/signal/PwmSignal.java
package tk.hiddenname.smarthome.model.signal;

import com.fasterxml.jackson.databind.PropertyNamingStrategy;
import com.fasterxml.jackson.databind.annotation.JsonNaming;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.ToString;

import javax.validation.constraints.NotNull;

@EqualsAndHashCode(callSuper = true)
@ToString(of = {"pwmSignal"}, callSuper = true)
@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
public class PwmSignal extends Signal {

    @NotNull
    @Getter
    public final int pwmSignal;

    public PwmSignal(Long id, Integer pwmSignal) {
        super(id);
        this.pwmSignal = pwmSignal;
    }
}

File: src/main/kotlin/tk/hiddenname/smarthome/model/signal/PwmSignal.kt
package tk.hiddenname.smarthome.model.signal

import com.fasterxml.jackson.annotation.JsonPropertyOrder
import com.fasterxml.jackson.databind.PropertyNamingStrategy.SnakeCaseStrategy
import com.fasterxml.jackson.databind.annotation.JsonNaming
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

@JsonNaming(SnakeCaseStrategy::class)
@JsonPropertyOrder(value = ["hardwareId", "pwmSignal"])
class PwmSignal(
    @field:Max(1024)
    @field:Min(0)
    @field:NotNull(message = "pwm signal should be specified")
    val pwmSignal: Int? = null
) : Signal() {

    constructor (hardwareId: Long, pwmSignal: Int?) : this(pwmSignal) {
        super.hardwareId = hardwareId
    }
}